ipmininet/router/config/bgp.py

```diff
@@ -488,7 +493,7 @@ class Peer:
     def __init__(self, base: 'Router', node: str, v6=False):
         """:param base: The base router that has this peer
         :param node: The actual peer"""
-        self.peer, other = self._find_peer_address(base, node, v6=v6)
+        self.peer, other, self.local_addr = self._find_peer_address(base, node, v6=v6)
```

Comment:
This is not always true. If BGP peers are not directly connected, the ```_find_peer_address``` does not find the local interface that is the closest to the peer. For that, you need to store it in the starting interface ```prio_queue``` and return it in the end

Reply:
I think that was changed in 53a7cd5224f7b281dabbd3fefafa6ba906b01749. But anyway, this is modified in 9af8167c1c77825d5e1d14ca1e89318c727975e8.  I had to slightly change the priority queue format because `IPIntf` does not override the `__lt__` method